rust/src/accountant.rs

```diff
@@ -0,0 +1,166 @@
+use crate::accumulator::{UsageAccumulator, UsageUnit};
+use crate::producer::{ClientError, KafkaConfig, KafkaProducer, Producer};
+use chrono::{Duration, Local};
+use serde_json::json;
+
+static DEFAULT_TOPIC_NAME: &str = "shared-resources-usage";
+/// This is the entry point for the library. It is in most cases
+/// everything you need to instrument your application.
+///
+/// The UsageAccountant needs a `Producer` and a `UsageAccumulator`.
+/// Data is stored in the accumulator and, periodically, it is
+/// flushed into the Kafka topic via a consumer.
+///
+/// Accumulating data locally is critical to reduce the performance impact
+/// of this library to a minimum and reduce the amount of Kafka messages.
+/// This means that this structure should be instantiated rarely.
+/// Possibly only once per application (or per thread).
+///
+/// Avoid creating a UsageAccountant every time some data needs to
+/// be recorded.
+
+pub struct UsageAccountant<'a> {
+    accumulator: UsageAccumulator,
+    producer: Box<dyn Producer + 'a>,
+    topic: String,
+}
+
+impl<'a> UsageAccountant<'a> {
+    /// Instantiates a UsageAccountant from a Kafka config object.
+    /// This initialization method lets the `UsageAccountant` create
+    /// the producer and own it.
+    ///
+    /// You should very rarely change topic name or granularity.
+    pub fn new(
+        producer_config: KafkaConfig,
+        topic_name: Option<&str>,
+        granularity_sec: Option<Duration>,
```

Comment:
Should this just be `granularity` since it's a Duration?

Reply:
oh right. Yeah now that it is not an integer anymore, no need to specify sec.

Reply:
done